ITEvents/Screens/Search tab/SearchViewController.swift

```diff
@@ -1,34 +1,68 @@
-//
-//  SearchViewController.swift
-//  ITEvents
-//
-//  Created by Sofia on 28/01/2018.
-//  Copyright © 2018 Sofia. All rights reserved.
-//
-
 import UIKit
+import DisplaySwitcher
+import Reusable
 
-class SearchViewController: UIViewController {
-
+class SearchViewController: UIViewController, ISearchView {
+    @IBOutlet private weak var searchBar: UISearchBar!
+    @IBOutlet private weak var collectionView: UICollectionView!
+    
+    var presenter: ISearchPresenter!
+    private var events = [EventCollectionCellViewModel]()
+    
+    private var layout: DisplaySwitchLayout!
+    private var layoutState: LayoutState!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        // Do any additional setup after loading the view.
+        registerNibs()
+        presenter.setup()
     }
-
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        collectionView.isUserInteractionEnabled = true
     }
     
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destinationViewController.
-        // Pass the selected object to the new view controller.
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        presenter.setupBeforeViewAppear(then: { self.collectionView.reloadData() })
```

Comment:
May be we can do it only when Search tab bar item became active.
Could you please investigate that option?